### PR TITLE
feat: add title on pins

### PR DESCRIPTION
**Pull request by ardasnturk**
- added the `Callout` component of the `react-native-maps` package to add title on the pins in the report screen
- added `calloutTextEnabled` prop to enable the `Callout` feature on the pin
- added condition to `onMapPress` prop in `sueReportLocation` to ensure that the location is not selected after clicking the pin on the map

SUE-75

## Screenshots:

|before|after|
|--|--|
Simulator Screenshot - iPhone 15 Pro - 2024-05-07 at 16 11 49|Simulator Screenshot - iPhone 15 Pro - 2024-05-07 at 16 11 46

### CODE REVIEW

File: src/components/map/Map.tsx [hunk 1]
```diff
@@ -191,6 +193,7 @@ export const Map = ({
         )}
         {locations?.map((marker, index) => {
           const isActiveMarker = selectedMarker && marker.id === selectedMarker;
+          const calloutText = marker.title || undefined;
```
Comment (donni106) [inline src/components/map/Map.tsx:196]: `|| undefined` is not necessary. `marker.title` returns `undefined` if it is not defined.

**Review by donni106 — APPROVED** [review]

File: src/components/map/Map.tsx [hunk 2]
```diff
@@ -246,6 +249,14 @@ export const Map = ({
                   iconName={marker.iconName ? marker.iconName : undefined}
                 />
               )}
+
+              {calloutTextEnabled && (
+                <Callout style={styles.callout}>
+                  <RegularText smallest center>
+                    {calloutText}
```
Comment (donni106) [inline src/components/map/Map.tsx:256]: what about directly `marker.title` instead of creating the variable `calloutText`?

Comment (ardasnturk) [reply src/components/map/Map.tsx:256]: Yes, very logical. I've made this change.
https://github.com/smart-village-solutions/smart-village-app-app/commit/219b63c72de3f54270f773026c7b5fa3318b29c8

Comment (donni106) [reply src/components/map/Map.tsx:256]: but now you have added an additional check, which needs !!

**Pull request merged**